src/liboslexec/backendllvm.cpp

```diff
@@ -590,7 +591,7 @@ BackendLLVM::llvm_load_value(llvm::Value* ptr, const TypeSpec& type, int deriv,
         ptr = ll.GEP(ptr, 0, component);
 
     // Now grab the value
-    llvm::Value* result = ll.op_load(ptr);
+    llvm::Value* result = ll.op_load(ptr, llname);
 
```

Comment:
For these more complicated entry points, I'm wondering if it would be more useful to bake things like deriv, arrayindex, component into the name so they are there in a consistent way.  Also, naming the intermediate results, such as the offset here so they are easily tied together as optimizations shuffle them around may also be helpful.

Reply:
LLVM automatically appends an incrementing counter to the string, if it is not unique.

How these are used will be a little more clear in my next PR, which is an overhaul of how strings are handled on GPU (and makes use of this). This is just setting up some of the debugging infrastructure that looked like I could cleave it off into a separate self-contained PR. I've just scratched the surface, I think this will continue to evolve as we realize what information is useful to embed in the names.

Reply:
@sfriedmapixar I definitely will extend to wide ops and maybe universally, but not as part of this first PR. I need to put the string stuff to bed first.

Reply:
Sounds completely resonable.